### PR TITLE
resource/firewall_rule: StateFunc to trim trailing space

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #105

```
▶ acctests triton TestAccTritonFirewallRule_
=== RUN   TestAccTritonFirewallRule_basic
--- PASS: TestAccTritonFirewallRule_basic (9.02s)
=== RUN   TestAccTritonFirewallRule_update
--- PASS: TestAccTritonFirewallRule_update (15.41s)
=== RUN   TestAccTritonFirewallRule_enable
--- PASS: TestAccTritonFirewallRule_enable (15.28s)
=== RUN   TestAccTritonFirewallRule_heredoc
--- PASS: TestAccTritonFirewallRule_heredoc (7.41s)
PASS
ok  	github.com/terraform-providers/terraform-provider-triton/triton	47.133s
```

### CODE REVIEW

**Comment by stack72** [comment]
@cheapRoc / @kwilczynski

File: triton/resource_firewall_rule.go [hunk 1]
```diff
@@ -23,6 +25,14 @@ func resourceFirewallRule() *schema.Resource {
 				Description: "firewall rule text",
 				Type:        schema.TypeString,
 				Required:    true,
+				StateFunc: func(v interface{}) string {
+					switch v.(type) {
+					case string:
+						return strings.TrimSpace(v.(string))
+					default:
+						return ""
+					}
```
Comment (tombuildsstuff) [inline triton/resource_firewall_rule.go]: If we're doing this, is it also worth trimming the space before sending this to the API?

Comment (cheapRoc) [reply triton/resource_firewall_rule.go]: Wondering if this should be ported into `triton-go` as well.

Comment (stack72) [reply triton/resource_firewall_rule.go]: @cheapRoc / @tombuildsstuff Triton API itself already strips the whitespace - so if someone specifies one in the TF config then we get the perpetual diff only

Comment (cheapRoc) [reply triton/resource_firewall_rule.go]: Fair enough.

**Pull request merged**